### PR TITLE
Packaging for npmjs (#11)

### LINKED ISSUE #11: NPM Packaging
We want to offer OpenIbex as NPM-Installable packages for everyone. 

We shall set up 
- Proper packaging
- CI-Pipeline, where tags can be published to NPM (preferrably automatically

### PR DESCRIPTION
**Pull request by tbergmueller**
This resolves #11 

- Packaging for npmjs of `@openibex/[core, cli, ethereum, chain]`
- This corresponds to the v0.0.4 already pushed to npmjs
-- https://www.npmjs.com/package/@openibex/core
-- https://www.npmjs.com/package/@openibex/chain
-- https://www.npmjs.com/package/@openibex/ethereum
-- https://www.npmjs.com/package/@openibex/cli

(Unrelated but related)
- A boilerplate-project using the already published packages is available at https://github.com/tbergmueller/openibex-boilerplate
-- Refer it's README for instructions
-- This demonstrates setup, init, read / write blockchain, subscribe events

Note this PR does not fix all issues code-wise, several where equipped with a `//@ts-ignore` as well as a `//FIXME @Lukas`. These need fixing: #13

### CODE REVIEW

File: packages/chain/src/indexer/indexer.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class OiEventIndexer{
   protected bloomFilters?: any;
   protected callback: (...args: any[]) => Promise<void>;
   
-  protected processedDB: OiNKeyValue<number>;
+  protected processedDB: OiNKeyValue<string>;
```
Comment (LukOfDecen) [inline packages/chain/src/indexer/indexer.ts:27]: Was a number previously (amount of events per block), then added the processing node Id. As every block is only processed by one node (and after that validated by multiple).

I probably need to check my compiler settings. Mean "weak typing ok at the moment", but wrong typing shall never pass.

File: packages/chain/src/indexer/indexer.ts [hunk 2]
```diff
@@ -48,19 +48,29 @@ export class OiEventIndexer{
...
 
+    // FIXME @Lukas Argument of type '(peerId: any) => Promise<void>' is not assignable to parameter of type 'never'.
+    // @ts-ignore
     this.processedDB.events.on('leave', async (peerId) => {
       delete(this.dbPeers[peerId]);
     });
 
-    const returnValue = await this.processedDB.newest();
-    const key = (returnValue === undefined) ? 0 : returnValue.key;
+    // FIXME @Lukas
```
Comment (LukOfDecen) [inline packages/chain/src/indexer/indexer.ts:63]: Will be fixed, start block logic is as follows:

- If a contract is indexed, deploy block of contract (algo still in works)
- If an account is filtered, the first block of account activity (balance).
- Inside a protocol, every contract on every chain has a different start block by default.

**Review by IbexLabsLuk — COMMENTED** [review]
IMO Ready to merge. (However: Due to time constraints did not checkout and start)

File: packages/chain/src/indexer/indexer.ts (hunk 1 above)
Comment (tbergmueller) [reply packages/chain/src/indexer/indexer.ts:27]: This PR will change compiler settings then anyway, we just need to adapt tsconfig.json and remove the `// @ts-ignore` - see #13

File: packages/chain/src/indexer/indexer.ts (hunk 2 above)
Comment (tbergmueller) [reply packages/chain/src/indexer/indexer.ts:63]: thx!

**Pull request merged**